sklego/preprocessing/patsytransformer.py

```diff
@@ -24,7 +24,12 @@ def fit(self, X, y=None):
 
         # check the number of observations hasn't changed. This ought not to
         # be necessary given NA_action='raise' above but just to be safe
-        assert np.array(X_).shape[0] == np.array(X).shape[0]
```

Comment:
`assert` should be avoided in production code as running non-debug mode would skip that anyway